util/src/main/java/io/kubernetes/client/util/credentials/TokenFileAuthentication.java

```diff
@@ -26,49 +27,23 @@
 // TODO: prefer OpenAPI backed Auentication once it is available. see details in
 // https://github.com/OpenAPITools/openapi-generator/pull/6036. currently, the
 // workaround is to hijack the http request.
-public class TokenFileAuthentication implements Authentication, Interceptor {
-  private String file;
-  private String token;
-  private Instant expiry;
-
+public class TokenFileAuthentication extends RefreshAuthentication {
   public TokenFileAuthentication(String file) {
-    this.expiry = Instant.MIN;
-    this.file = file;
-  }
-
-  private String getToken() {
-    if (Instant.now().isAfter(this.expiry)) {
-      try {
-        this.token =
-            new String(Files.readAllBytes(Paths.get(this.file)), Charset.defaultCharset()).trim();
-        expiry = Instant.now().plusSeconds(60);
-      } catch (IOException ie) {
-        throw new RuntimeException("Cannot read file: " + this.file);
-      }
-    }
-
-    return this.token;
+    this(file, Duration.ofMinutes(1));
   }
 
-  public void setExpiry(Instant expiry) {
-    this.expiry = expiry;
+  public TokenFileAuthentication(String file, Duration refreshPeriod) {
+    super(() -> {
+      return getToken(file);
+    }, refreshPeriod);
   }
 
-  public void setFile(String file) {
-    this.file = file;
-  }
-
-  @Override
-  public void provide(ApiClient client) {
-    OkHttpClient withInterceptor = client.getHttpClient().newBuilder().addInterceptor(this).build();
-    client.setHttpClient(withInterceptor);
+  private static String getToken(String file) {
+    try {
+      return new String(Files.readAllBytes(Paths.get(file)), Charset.defaultCharset()).trim();
+    } catch (IOException e) {
+      throw new RuntimeException("Cannot read file: " + file);
```

Comment:
We should probably convert this to some sort of non-RuntimeException like "FailedTokenAquisitionException" and handle that inside the `RefreshAuthentication.java`

Reply:
That sounds like a good idea but I ended up adding the exception conversion as a runtime/unchecked exception in the parent class `RefreshAuthentication`. I also tried making it a checked exception in the token supplier function however it will break backward compatibility so I implemented the unchecked exception in the new commit. one example that compatibility would be broken is here where we're loading the token eagerly in `RefreshAuthentication` :

> https://github.com/kubernetes-client/java/blob/master/util/src/main/java/io/kubernetes/client/util/credentials/RefreshAuthentication.java#L43

so the new checked exception would have to be added to the constructor which will require all places referencing it adding explicit handling the checked exception. does that make sense?